readthedocs/v3/serializers.py

```diff
@@ -0,0 +1,444 @@
+        path = reverse(
+            'projects-versions-list',
+            kwargs={
+                'parent_lookup_project__slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_builds(self, obj):
+        path = reverse(
+            'projects-builds-list',
+            kwargs={
+                'parent_lookup_project__slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_subprojects(self, obj):
+        path = reverse(
+            'projects-subprojects',
+            kwargs={
+                'project_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_superproject(self, obj):
+        path = reverse(
+            'projects-superproject',
+            kwargs={
+                'project_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_translations(self, obj):
+        path = reverse(
+            'projects-translations',
+            kwargs={
+                'project_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+
+class ProjectSerializer(FlexFieldsModelSerializer):
+
+    language = LanguageSerializer()
+    programming_language = ProgrammingLanguageSerializer()
+    repository = RepositorySerializer(source='*')
+    privacy_level = PrivacyLevelSerializer(source='*')
+    urls = ProjectURLsSerializer(source='*')
+    subproject_of = serializers.SerializerMethodField()
+    translation_of = serializers.SerializerMethodField()
+    default_branch = serializers.CharField(source='get_default_branch')
+    tags = serializers.StringRelatedField(many=True)
+
+    description = serializers.SerializerMethodField()
+
+    links = ProjectLinksSerializer(source='*')
+
+    # TODO: adapt these fields with the proper names in the db and then remove
+    # them from here
+    created = serializers.DateTimeField(source='pub_date')
+    modified = serializers.DateTimeField(source='modified_date')
+
+    expandable_fields = dict(
+        users=(
+            UserSerializer, dict(
+                source='users',
+                many=True,
+            ),
+        ),
+        active_versions=(
+            VersionSerializer, dict(
+                # NOTE: this has to be a Model method, can't be a
+                # ``SerializerMethodField`` as far as I know
+                source='active_versions',
+                many=True,
+            ),
+        ),
+    )
+
+    class Meta:
+        model = Project
+        fields = [
+            'id',
+            'name',
+            'slug',
+            'description',
+            'created',
+            'modified',
+            'language',
+            'programming_language',
+            'repository',
+            'default_version',
+            'default_branch',
+            'privacy_level',
+            'subproject_of',
+            'translation_of',
+            'urls',
```

Comment:
This field (via our resolver - see #3712) results in two extra queries per project returned.

Reply:
The problem with this `urls` field is the usage of `Project.get_docs_url` method. I need to check how to improve this.